frontend/simple/js/state.js

```diff
@@ -92,6 +92,26 @@ const getters = {
 }
 
 const actions = {
+  async login (
+    {dispatch, commit, state}: {dispatch: Function, commit: Function, state: Object},
+    user: string
+  ) {
+    commit('login', user)
+    await dispatch('loadSettings')
+    await db.saveCurrentUser(user)
+    for (let [key] of Object.entries(state.contracts)) {
```

Comment:
If you just want the key, better to use the `Object.keys` method, since `Object.entries` is idiomatically used for fetching both the key and value at the same time.